packages/manager/modules/pci/src/projects/project/notebooks/notebooks.routing.js

```diff
@@ -218,10 +218,12 @@ export default /* @ngInject */ ($stateProvider) => {
       ) => () => {
         notebooks.forEach((notebook) => {
           if (notebook.isPending()) {
-            NotebookService.pollNotebookStatus(
-              projectId,
-              notebook.id,
-            ).then((notebookInfo) => notebook.updateData(notebookInfo));
+            NotebookService.pollNotebookStatus(projectId, notebook.id).then(
+              // success function
+              (notebookInfo) => notebook.updateData(notebookInfo),
+              // if 404, fall here and delete notebook
```

Comment:
**question:** Couldn't we use a `.catch()`?

Reply:
The ```pollNotebookStatus``` uses a ```Poller```. According to[ the documentation](https://www.npmjs.com/package/@ovh-ux/ng-ovh-swimming-poll) of @ovh-ux/ng-ovh-swimming-poll, the error state is handled in the second arrow function in the ```.then()``` :
```
Poller.poll(
    url,
    null, // params
    {
      namespace: 'a_namespace',
      method: 'get',
      scope: $scope.$id,
    },
  ).then(
    (result) => {
      console.log('result contains http response if task status is successful');
    },
    (result) => {
      console.log(
        'result contains http response if task status is in error state',
      );
    },
    (result) => {
      console.log(
        'result contains http response if task status is in pending state',
      );
    },
  );
```